cartographer/mapping/global_trajectory_builder.h

```diff
@@ -29,12 +29,14 @@ template <typename LocalTrajectoryBuilder,
 class GlobalTrajectoryBuilder
     : public mapping::GlobalTrajectoryBuilderInterface {
  public:
-  GlobalTrajectoryBuilder(const LocalTrajectoryBuilderOptions& options,
-                          const int trajectory_id,
-                          SparsePoseGraph* const sparse_pose_graph)
+  GlobalTrajectoryBuilder(
+      const LocalTrajectoryBuilderOptions& options, const int trajectory_id,
+      SparsePoseGraph* const sparse_pose_graph,
+      InsertedRangeDataCallback* const inserted_range_data_callback)
```

Comment:
Passing a pointer seems like premature optimization. Pass a copy?

Reply:
Same as below - what if we did not provide a callback, which is covered by passing a null pointer?

Reply:
You can also compare a `std::function<>` as a bool. It compares false if it is default constructed. `if (the_function) { ... }`.